gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BigQuery.java

```diff
@@ -25,5 +34,602 @@
  */
 public interface BigQuery extends Service<BigQueryOptions> {
 
-  // TODO(mziccard) add missing methods
+  enum DatasetField {
```

Comment:
We don't have it for storage (but this is no good reason for not adding it). I think that the names are quite self-explanatory. I also think that adding javadoc for each enum value would mean adding too much "obvious" comments. I'm +1 to adding a javadoc comment on top of the enum type with a link to a dataset/table/job resource definition.